### PR TITLE
♻️ api-server: openapi docs and versioning

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

None critical changes for the `api-server` to align with the new release strategy developed for the client in https://github.com/ITISFoundation/osparc-simcore-clients/pull/180

- 📝 new formatted messages for **changelog**
- ♻️ [**new version identifier**](https://packaging.python.org/en/latest/specifications/version-specifiers/#public-version-identifiers) if `API_SERVER_DEV_FEATURES_ENABLED=1`
      - the OAS exposes all entrypoint (some are hidden in production)
      - Changes to a development version `M.m.p.post0.dev`
      - The resulting OAS can be obtained also using `make openapi-dev.json` 
- 📝 makefile recipes to produce OAS diffs between different versions

## Related issue/s

- Part of ITISFoundation/osparc-issues#1611

## Dev-ops checklist

None

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks!

File: services/api-server/src/simcore_service_api_server/api/routes/_constants.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# CHANGELOG formatted-messages for API routes
+#
+# - Append at the bottom of the route's description
+# - These are displayed in the swagger doc
+# - These are displayed in client's doc as well (auto-generator)
+# - Inspired on this idea https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#describing-changes-between-versions
+#
+
+# new routes
+FMSG_CHANGELOG_NEW_IN_VERSION: Final[str] = "New in *version {}*\n"
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/_constants.py:13]: what means FMSG?

File: services/api-server/src/simcore_service_api_server/api/routes/_constants.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# new inputs/outputs in routes
+FMSG_CHANGELOG_ADDED_IN_VERSION: Final[str] = "Added in *version {}*: {}\n"
+
+# changes on inputs/outputs in routes
+FMSG_CHANGELOG_CHANGED_IN_VERSION: Final[str] = "Changed in in *version {}*: {}\n"
+
+# removed on inputs/outputs in routes
+FMSG_CHANGELOG_REMOVED_IN_VERSION_FORMAT: Final[
+    str
+] = "Removed in in *version {}*: {}\n"
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/_constants.py:24]:
```suggestion
] = "Removed in *version {}*: {}\n"
```

File: services/api-server/src/simcore_service_api_server/api/routes/_constants.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#
+
+# new routes
+FMSG_CHANGELOG_NEW_IN_VERSION: Final[str] = "New in *version {}*\n"
+
+# new inputs/outputs in routes
+FMSG_CHANGELOG_ADDED_IN_VERSION: Final[str] = "Added in *version {}*: {}\n"
+
+# changes on inputs/outputs in routes
+FMSG_CHANGELOG_CHANGED_IN_VERSION: Final[str] = "Changed in in *version {}*: {}\n"
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/api/routes/_constants.py:19]:
```suggestion
FMSG_CHANGELOG_CHANGED_IN_VERSION: Final[str] = "Changed in *version {}*: {}\n"
```

File: services/api-server/src/simcore_service_api_server/core/application.py [hunk 4]
```diff
@@ -20,17 +21,24 @@
...
 
-def _label_info_with_state(settings: ApplicationSettings, title: str, version: str):
+def _label_title_and_version(settings: ApplicationSettings, title: str, version: str):
     labels = []
     if settings.API_SERVER_DEV_FEATURES_ENABLED:
-        labels.append("dev")
+        # builds public version identifier with pre: `[N!]N(.N)*[{a|b|rc}N][.postN][.devN]`
+        # SEE https://packaging.python.org/en/latest/specifications/version-specifiers/#public-version-identifiers
+        v = Version(version)
+        version = f"{v.base_version}.post0.dev"
```
Comment (sanderegg) [inline services/api-server/src/simcore_service_api_server/core/application.py:30]: :aaa: ;) cool

**Review by sanderegg — APPROVED** [review]
thanks

File: services/api-server/src/simcore_service_api_server/api/routes/_constants.py (hunk 1 above)
Comment (pcrespov) [reply services/api-server/src/simcore_service_api_server/api/routes/_constants.py:13]: the concept of `f-string`  we know, right?
These as messages i.e. MSG and `FMSG` suggest `f-messages` or formatted messages

**Pull request merged**